cpg-language-python/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/python/ExpressionHandler.kt

```diff
@@ -99,13 +99,25 @@ class ExpressionHandler(frontend: PythonLanguageFrontend) :
         }
     }
 
+    /**
+     * Works similar to [apply] but before executing [block], it enters the scope for this object
+     * and afterward leaves the scope again.
+     */
+    private inline fun <T : Node> T.applyWithScope(block: T.() -> Unit): T {
```

Comment:
This could be a helpful bit, also for the future node builder API, should we move this to the scope manager? Or keep it intentionally private for now?

Reply:
I thought the same and wasn't sure where would be a good place. Maybe `Node` directly?

Reply:
When in doubt place it in `Extensions.kt`:D